### PR TITLE
CLN: replace %s syntax with .format in pandas.core.indexes

### PR DESCRIPTION
**Pull request by makeajourney**
progress towards #16130

- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by jreback** [comment]
this has some failures (ignore the 3.7 numpy_dev ones for now), but the others are legit

**Comment by jreback** [comment]
can you merge master

File: pandas/core/indexes/frozen.py [hunk 1]
```diff
@@ -113,8 +113,8 @@ def __unicode__(self):
                             escape_chars=('\t', '\r', '\n'))
 
     def __repr__(self):
-        return "%s(%s)" % (self.__class__.__name__,
-                           str(self))
+        return "{class_name}({list!s})".format(
+            class_name=self.__class__.__name__, list=self)
```
Comment (jbrockmendel) [inline pandas/core/indexes/frozen.py]: can you use a kwarg other than `list` here to avoid overlap with builtin name

**Comment by jreback** [comment]
closing as stale; ping if you want to continue

**Pull request closed without merging**